cmake/everest-generate.cmake

```diff
@@ -26,13 +26,7 @@ set_target_properties(generate_cpp_files
 #
 # out-of-tree interfaces/types/modules support
 #
-function(ev_add_project)
-    # FIXME (aw): resort to proper argument handling!
-    if (ARGC EQUAL 2)
-        set (EVEREST_PROJECT_DIR ${ARGV0})
-        set (EVEREST_PROJECT_NAME ${ARGV1})
```

Comment:
Why can those args be omitted?

Reply:
I can re-add them but I've not seen them used anywhere

Reply:
oh nvmd, I have overseen that there are set default values.

So the following can be shortened a bit, since with removing the args they are definetly not defined:
```
    if (NOT EVEREST_PROJECT_DIR)
        # if we don't get a directory, we're assuming project directory
        set (EVEREST_PROJECT_DIR ${PROJECT_SOURCE_DIR})
        set (CALLED_FROM_WITHIN_PROJECT TRUE)
    elseif (NOT EXISTS ${EVEREST_PROJECT_DIR})
        message(FATAL_ERROR "${CMAKE_CURRENT_FUNCTION} got non-existing project path: ${EVEREST_PROJECT_DIR}")
    endif ()
    if (NOT EVEREST_PROJECT_NAME)
        set (EVEREST_PROJECT_NAME ${PROJECT_NAME})
    endif ()
```

Reply:
I've readded them, they were needed 😅